File: GithubUserJava/app/src/main/java/com/example/myapplication/factory/FollowViewModelFactory.java
package com.example.myapplication.factory;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import com.example.myapplication.viewmodel.FollowViewModel;

public class FollowViewModelFactory implements ViewModelProvider.Factory {

    private final String username;

    public FollowViewModelFactory(String username) {
        this.username = username;
    }

    @SuppressWarnings("unchecked")
    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        if (modelClass.isAssignableFrom(FollowViewModel.class)) {
            return (T) new FollowViewModel(username);
        }
        throw new IllegalArgumentException("Unknown ViewModel class");
    }
}

File: GithubUserKotlin/app/src/main/java/com/example/myapplication/factory/FollowViewModelFactory.kt
package com.example.myapplication.factory

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.example.myapplication.viewmodel.FollowViewModel

class FollowViewModelFactory(private val username: String) : ViewModelProvider.Factory {
    @Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(FollowViewModel::class.java)) {
            return FollowViewModel(username) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }

}